libfreerdp/core/gcc.c

```diff
@@ -885,18 +999,11 @@ static BOOL updateEarlyClientCaps(rdpSettings* settings, UINT32 earlyCapabilityF
 		settings->SupportDynamicTimeZone =
 		    (earlyCapabilityFlags & RNS_UD_CS_SUPPORT_DYNAMIC_TIME_ZONE) ? TRUE : FALSE;
 
-	if (settings->SupportStatusInfoPdu)
-		settings->SupportStatusInfoPdu =
-		    (earlyCapabilityFlags & RNS_UD_CS_SUPPORT_STATUSINFO_PDU) ? TRUE : FALSE;
-
-	if (settings->SupportErrorInfoPdu)
-		settings->SupportErrorInfoPdu =
-		    earlyCapabilityFlags & RNS_UD_CS_SUPPORT_ERRINFO_PDU ? TRUE : FALSE;
-
-	if (!(earlyCapabilityFlags & RNS_UD_CS_VALID_CONNECTION_TYPE))
+	if ((earlyCapabilityFlags & RNS_UD_CS_VALID_CONNECTION_TYPE) == 0)
 		connectionType = 0;
 	settings->ConnectionType = connectionType;
 
+	filterAndLogEarlyClientCapabilityFlags(earlyCapabilityFlags);
```

Comment:
This call seems redundant (the returned, filtered value is not used and logging of the flags is already done by the caller)

Reply:
yes, the returned value is intendendly unused, but the logging here is different. while the caller just does some debug level logging the filter warns about the flags, which is the intention of this line.